qiskit/primitives/backend_sampler.py

```diff
@@ -151,15 +151,29 @@ def _call(
         bound_circuits = self._bound_pass_manager_run(bound_circuits)
 
         # Run
-        result = self._backend.run(bound_circuits, **run_options).result()
+        if isinstance(self._backend, BackendV1):
```

Comment:
It might be simpler to call `_run_circuits` of Estimator?

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8955/commits/dda0f0607e3404e6d9392806bd3946b62bec5f9e